some warnings

There are some warnings I would like to clear up. I hates warnings!!

```
ibtool: compile:  mpicc -DHAVE_CONFIG_H -I. -I../../../src/clib -I../.. -I/usr/include -std=c99 -MT pio_spmd.lo -MD -MP -MF .deps/pio_spmd.Tpo -c ../../../src/clib/pio_spmd.c  -fPIC -DPIC -o .libs/pio_spmd.o
../../../src/clib/pioc.c: In function ‘PIOc_set_hint’:
../../../src/clib/pioc.c:1054:9: warning: passing argument 2 of ‘MPI_Info_set’ discards ‘const’ qualifier from pointer target type [enabled by default]
         if ((mpierr = MPI_Info_set(ios->info, hint, hintval)))
         ^
In file included from ../../../src/clib/pio.h:16:0,
                 from ../../../src/clib/pioc.c:11:
/usr/lib/openmpi/include/mpi.h:1297:20: note: expected ‘char *’ but argument is of type ‘const char *’
 OMPI_DECLSPEC  int MPI_Info_set(MPI_Info info, char *key, char *value);
                    ^
../../../src/clib/pioc.c:1054:9: warning: passing argument 3 of ‘MPI_Info_set’ discards ‘const’ qualifier from pointer target type [enabled by default]
         if ((mpierr = MPI_Info_set(ios->info, hint, hintval)))
         ^
In file included from ../../../src/clib/pio.h:16:0,
                 from ../../../src/clib/pioc.c:11:
/usr/lib/openmpi/include/mpi.h:1297:20: note: expected ‘char *’ but argument is of type ‘const char *’
 OMPI_DECLSPEC  int MPI_Info_set(MPI_Info info, char *key, char *value);
```
and:

```
mpicc -DHAVE_CONFIG_H -I. -I../../../tests/cunit -I../..  -I../../../src/clib -I/usr/include  -std=c99 -MT test_darray_multi.o -MD -MP -MF .deps/test_darray_multi.Tpo -c -o test_darray_multi.o ../../../tests/cunit/test_darray_multi.c
../../../tests/cunit/test_async_mpi.c:137:24: warning: format ‘%d’ expects argument of type ‘int’, but argument 5 has type ‘MPI_Comm’ [-Wformat=]
                        my_rank, cmp, comproot[cmp], union_comm[cmp]);
                        ^
../../../tests/cunit/test_async_mpi.c:142:17: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘MPI_Request’ [-Wformat=]
                 printf("my_rank %d MPI_Irecv req[%d] = %d\n", my_rank, cmp, req[cmp]);
                 ^
../../../tests/cunit/test_async_mpi.c:160:24: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘MPI_Request’ [-Wformat=]
                        my_rank, req[0], MPI_REQUEST_NULL);
                        ^
../../../tests/cunit/test_async_mpi.c:160:24: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘struct ompi_request_t *’ [-Wformat=]
../../../tests/cunit/test_async_mpi.c:162:21: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘MPI_Request’ [-Wformat=]
                     printf("my_rank %d req[%d] = %d\n", my_rank, c, req[c]);
                     ^
../../../tests/cunit/test_async_mpi.c:167:17: warning: format ‘%d’ expects argument of type ‘int’, but argument 5 has type ‘MPI_Request’ [-Wformat=]
                 printf("my_rank %d Waitany returned index = %d req[%d] = %d\n", my_rank, index, index, req[index]);
                 ^
../../../tests/cunit/test_async_mpi.c:173:20: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘MPI_Comm’ [-Wformat=]
                    index, msg);
                    ^
../../../tests/cunit/test_async_mpi.c:180:20: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘MPI_Comm’ [-Wformat=]
                    index, msg);
                    ^
mpicc -DHAVE_CONFIG_H -I. -I../../../tests/cunit -I../..  -I../../../src/clib -I/usr/include  -std=c99 -MT test_darray_multivar.o -MD -MP -MF .deps/test_darray_multivar.Tpo -c -o test_darray_multivar.o ../../../tests/cunit/test_darray_multivar.c
../../../tests/cunit/test_async_mpi.c:200:24: warning: format ‘%d’ expects argument of type ‘int’, but argument 5 has type ‘MPI_Comm’ [-Wformat=]
                        my_rank, index, comproot[index], union_comm[index]);
                        ^
../../../tests/cunit/test_async_mpi.c:205:17: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘MPI_Request’ [-Wformat=]
                 printf("my_rank %d msg_handler called MPI_Irecv req[%d] = %d\n", my_rank, index, req[index]);
                 ^
../../../tests/cunit/test_async_mpi.c: In function ‘main’:
../../../tests/cunit/test_async_mpi.c:277:13: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘struct ompi_group_t *’ [-Wformat=]
             printf("MPI_GROUP_NULL %d MPI_COMM_NULL %d\n", MPI_GROUP_NULL, MPI_COMM_NULL);
             ^
../../../tests/cunit/test_async_mpi.c:277:13: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘struct ompi_communicator_t *’ [-Wformat=]
mpicc -DHAVE_CONFIG_H -I. -I../../../tests/cunit -I../..  -I../../../src/clib -I/usr/include  -std=c99 -MT test_darray_multivar2.o -MD -MP -MF .deps/test_darray_multivar2.Tpo -c -o test_darray_multivar2.o ../../../tests/cunit/test_darray_multivar2.c
../../../tests/cunit/test_async_mpi.c:286:13: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘MPI_Comm’ [-Wformat=]
             printf("my_rank %d created io comm io_comm = %d\n", my_rank, io_comm);
             ^
../../../tests/cunit/test_async_mpi.c:320:17: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘MPI_Group’ [-Wformat=]
                 printf("my_rank %d created component MPI group - group[%d] = %d\n", my_rank, cmp, group[cmp]);
                 ^
../../../tests/cunit/test_async_mpi.c:351:24: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘MPI_Comm’ [-Wformat=]
                        my_rank, cmp, comp_comm[cmp], comp_rank);
                        ^
../../../tests/cunit/test_async_mpi.c:360:21: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘MPI_Comm’ [-Wformat=]
                     printf("my_rank %d dup of io_comm = %d io_rank = %d\n", my_rank, io_comm, io_rank);
                     ^
mpicc -DHAVE_CONFIG_H -I. -I../../../tests/cunit -I../..  -I../../../src/clib -I/usr/include  -std=c99 -MT test_darray_multivar3.o -MD -MP -MF .deps/test_darray_multivar3.Tpo -c -o test_darray_multivar3.o ../../../tests/cunit/test_darray_multivar3.c
../../../tests/cunit/test_async_mpi.c:372:24: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘MPI_Group’ [-Wformat=]
                        my_rank, cmp, union_group[cmp], nprocs_union, cmp, union_comm[cmp]);
                        ^
../../../tests/cunit/test_async_mpi.c:372:24: warning: format ‘%d’ expects argument of type ‘int’, but argument 7 has type ‘MPI_Comm’ [-Wformat=]
../../../tests/cunit/test_async_mpi.c:424:25: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘MPI_Comm’ [-Wformat=]
                         printf("my_rank %d sending exit message on union_comm %d\n", my_rank, union_comm[cmp]);
                         ^
```

More warnings:

```
pioc_support.c: In function ‘PIOc_readmap’:
pioc_support.c:880:9: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
         fscanf(fp,"version %d npes %d ndims %d\n",&rversno, &rnpes, ndims);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pioc_support.c:895:13: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
             fscanf(fp,"%d ", tdims + i);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
pioc_support.c:902:13: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
             fscanf(fp, "%d %lld", &j, &maplen);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pioc_support.c:908:17: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
                 fscanf(fp, "%lld ", tmap+j);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/clib/pioc_support.c
+++ b/src/clib/pioc_support.c
@@ -880,8 +880,8 @@ int PIOc_readmap(const char *file, int *ndims, int **gdims, PIO_Offset *fmaplen,
         if (!fp)
             pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
 
-        fscanf(fp,"version %d npes %d ndims %d\n",&rversno, &rnpes, ndims);
-
+        if (fscanf(fp,"version %d npes %d ndims %d\n", &rversno, &rnpes, ndims) != 3)
+            pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
         if (rversno != VERSNO)
             return pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
 
@@ -895,20 +895,23 @@ int PIOc_readmap(const char *file, int *ndims, int **gdims, PIO_Offset *fmaplen,
         if (!(tdims = calloc(*ndims, sizeof(int))))
             return pio_err(NULL, NULL, PIO_ENOMEM, __FILE__, __LINE__);
         for (int i = 0; i < *ndims; i++)
-            fscanf(fp,"%d ", tdims + i);
+            if (fscanf(fp,"%d ", tdims + i) != 1)
+                pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
 
         if ((mpierr = MPI_Bcast(tdims, *ndims, MPI_INT, 0, comm)))
             return check_mpi(NULL, mpierr, __FILE__, __LINE__);
 
         for (int i = 0; i < rnpes; i++)
         {
-            fscanf(fp, "%d %lld", &j, &maplen);
+            if (fscanf(fp, "%d %lld", &j, &maplen) != 2)
+                pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
             if (j != i)  // Not sure how this could be possible
                 return pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
             if (!(tmap = malloc(maplen * sizeof(PIO_Offset))))
                 return pio_err(NULL, NULL, PIO_ENOMEM, __FILE__, __LINE__);
             for (j = 0; j < maplen; j++)
-                fscanf(fp, "%lld ", tmap+j);
+                if (fscanf(fp, "%lld ", tmap + j) != 1)
+                    pio_err(NULL, NULL, PIO_EINVAL, __FILE__, __LINE__);
 
             if (i > 0)
             {

